### PR TITLE
Added buildFlat to @ballerine/blocks

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

___

### **PR Type**
enhancement

___

### **Description**
- Implemented a new `buildFlat` method in `BlocksBuilder` to simplify flattening of block structures.
- Added unit and integration tests for the `buildFlat` method.
- Updated type definitions in TypeScript files to use more concise array syntax.
- Updated package versions and dependencies in `backoffice-v2` and `kyb-app`.
- Updated changelogs and package versions in response to new features.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>blocks.spec-d.ts</strong><dd><code>Add tests for `buildFlat` and minor type adjustments</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/src/blocks.spec-d.ts
<li>Added a new test suite for <code>buildFlat</code> method.<br> <li> Adjusted the <code>debug</code> option in <code>createTestBlocks</code> to consider <br><code>process.env.CI</code>.<br> <li> Changed type of <code>value</code> in <code>TCell</code> from <code>Array<string></code> to <code>string[]</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-cf52e6bf0769d9c5af7b746c30a8daab72bec4506d80a55e01c84dbb2e3c24d2">+48/-2</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>blocks.spec.ts</strong><dd><code>Integration tests for `buildFlat` and type adjustments</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/src/blocks.spec.ts
<li>Added integration tests for the <code>buildFlat</code> method.<br> <li> Modified <code>debug</code> setting in <code>createTestBlocks</code> to be environment <br>sensitive.<br> <li> Updated type of <code>value</code> in <code>TCell</code> from <code>Array<string></code> to <code>string[]</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-59a960d67b0f53fd08650125068cbb5543191a9cd13a9d2489cede732a798060">+28/-2</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>blocks.ts</strong><dd><code>Implement `buildFlat` method and refactor types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/src/blocks.ts
<li>Introduced <code>buildFlat</code> method in <code>BlocksBuilder</code> class.<br> <li> Changed <code>Block</code> and <code>Blocks</code> types from <code>Array</code> to shorthand <code>[]</code> syntax.<br> <li> Updated several type definitions to use more concise array type <br>syntax.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-4c2f570d5eab669141f37f3ebd79c128444e138311b206180bf077b941baa0e1">+20/-12</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Documentation
</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog for backoffice-v2</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/CHANGELOG.md
<li>Added changelog entry for version 0.7.8 noting updated dependencies.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-ab29a72174aaef48b64727f8a8fcaa625cc8b1fd349f18c792c0360d9ac957c3">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog for kyb-app</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/kyb-app/CHANGELOG.md
<li>Added changelog entry for version 0.3.9 noting updated dependencies.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-4b4d337d8a6eeeef3b182a137501c54b32553c4b14b78e97832fad5c926f2d69">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog for blocks package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/CHANGELOG.md
<li>Added changelog entry for version 0.2.3 describing the addition of <br><code>buildFlat</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-88c9889d164b31c659049f4e6323f2c4d50c290c8b73b17dc0fa91102db8609e">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Bump backoffice-v2 version and update blocks dependency</code>&nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/package.json
<li>Updated version to 0.7.8.<br> <li> Updated dependency for <code>@ballerine/blocks</code> to version 0.2.3.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-69299682ca7fd0c3b4e72d9aad9ef3cb4bd7eff177034674b7d78e04c408191f">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Bump kyb-app version and update blocks dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/kyb-app/package.json
<li>Updated version to 0.3.9.<br> <li> Updated dependency for <code>@ballerine/blocks</code> to version 0.2.3.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-e5c452c604cb259ece186d5819d86f83025736d48fb666654b3dc429ec034cac">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.eslintrc.cjs</strong><dd><code>Update ESLint configuration in blocks package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/.eslintrc.cjs
<li>Added <code>parserOptions</code> with <code>project</code> set to <code>./tsconfig.eslint.json</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-3d877b09a8a0d7e4c9fa508b64fd6e49d98b6bc1cf1b5408f74e5ba8f551f797">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Update version in blocks package</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/blocks/package.json
- Updated version to 0.2.3.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-c2bb4b0e39839a40c8cbea9d0168a7b644e742760ec8a73fe4cdef6ce144cc0f">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pnpm-lock.yaml</strong><dd><code>Update dependencies in pnpm lock file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pnpm-lock.yaml
- Updated various dependencies to align with new versions.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2376/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

## Summary by CodeRabbit

- **New Features**
  - Introduced the `buildFlat` function in the `@ballerine/blocks` package for improved performance.

- **Updates**
  - Updated `@ballerine/blocks` dependency to version 0.2.3 in `backoffice-v2` and `kyb-app`.
  - Upgraded `backoffice-v2` to version 0.7.8 and `kyb-app` to version 0.3.9.

- **Bug Fixes**
  - Enhanced type definitions and testing configurations for better reliability and maintainability.

### CODE REVIEW

File: packages/blocks/.eslintrc.cjs [hunk 1]
```diff
@@ -4,4 +4,7 @@ module.exports = {
     browser: true,
   },
   extends: ['@ballerine/eslint-config'],
+  parserOptions: {
+    project: './tsconfig.eslint.json',
+  },
```
Comment (Omri-Levy) [inline packages/blocks/.eslintrc.cjs:9]: Fixes ESLint IDE error.

File: packages/blocks/src/blocks.spec-d.ts [hunk 2]
```diff
@@ -8,12 +8,12 @@ type TCell =
...
   | {
       type: 'headings';
-      value: Array<string>;
+      value: string[];
     };
 
 const createTestBlocks = () =>
   createBlocks<TCell>({
-    debug: true,
+    debug: !process.env.CI,
```
Comment (Omri-Levy) [inline packages/blocks/src/blocks.spec-d.ts:16]: Will reduce noise in CI tests.

Comment (tomer-shvadron) [reply packages/blocks/src/blocks.spec-d.ts:16]: Good job

**Review by tomer-shvadron — APPROVED** [review]
Awesome, I am just missing the refactor of `.build().flat()` to `.buildFlat()`

**Pull request merged**